### ISSUE TITLE
Talent templates keep disappearing randomly

### ISSUE THREAD

**Issue by Yeelbil** [Lethay/TalentedClassic#50]
Ive had to re-add my 3 templates like 10 times since i installed this addon, it seems like every time you close WoW it has a 50% chance of removing one of your templates. The most ive ran with no removal is 2 days, if i keep doing /reload it will eventually remove the templates.

Ive tried running ONLY this addon enabled and its still doing it.

Any more information i can provide? Would love to run this addon, but now i have to have the wowhead links to the specs saved and re-import them every other day..

**Comment by Lethay** [comment]
That's strange. That's never happened to me and no-one else has reported it. How long has it been happening for? Can you send me your settings file?

**Comment by Yeelbil** [comment]
> That's strange. That's never happened to me and no-one else has reported it. How long has it been happening for? Can you send me your settings file?

Yeah ive tried to look for anyone else with a similar issue but cant find anyone else, the setting are 100% default. I have not touched anything, reinstalled it multiple times.

**Comment by Lethay** [comment]
Could you send me your settings file as it is anyway? I'm thinking it could be

- Strangely invalid template?
- Unusual characters in your name or your template's names?

But I have no strong ideas.

**Comment by Yeelbil** [comment]
> Could you send me your settings file as it is anyway? I'm thinking it could be
> 
> * Strangely invalid template?
> * Unusual characters in your name or your template's names?
> 
> But I have no strong ideas.

Where can i find the settings file?

My Character name is as basic as it gets, 4 letters, no special characters. Ive tried with templates called just "1, 2, 3" but that resets them to.

**Comment by Lethay** [comment]
**WoW_install_location**/_classic_/WTF/Account/**(your-account-name)**/SavedVariables/Talented.lua

**Comment by Yeelbil** [comment]
> **WoW_install_location**/_classic_/WTF/Account/**(your-account-name)**/SavedVariables/Talented.lua

There is no such file there. Maybe we've found the issue?

**Comment by Lethay** [comment]
Maybe. Search for a file named Talented.lua within the WTF folder using your device's file browser, see if it's elsewhere.

**Comment by Yeelbil** [comment]
Found them now.. Maybe looked at the wrong place. [Here is the link to the files](https://drive.google.com/drive/folders/1Vozru_GY426hfdj8geCXmbVj9GY8OCAl?usp=sharing)

Just logged in to test and ive lost 2/3 talent templates i had set, one remaining. Did 4x reloads and then the last one got removed aswell.

**Comment by Lethay** [comment]
Hi,

Looking at this, I think it might be because you have Talented_Data disabled. But it's also interesting to note that it seems to delete templates when that addon is disabled -- it's probably deleting templates that belong to a class you're not logged in as, if my guess was right. Could you confirm this?

**Comment by Yeelbil** [comment]
The addon Talented_Data is not disabled in the game for me looking at the addons menu. 

If i now create a template on my main (Warrior) and log on my mage (Alt) its not deleting any templates as i test it now. The deletion is so random somtimes like i said in the first post, sometimes i can go a week and it not removing a template. What did remove all of my templates just now was i created 2 more templates, logged on my mage and warrior back and forth, templates still there.

Exit game, log back into my warrior, all templates gone.

**Comment by Lethay** [comment]
Could you turn the *force load* option on and see if the problem persists?

2021年7月25日(日) 16:01 Yeelbil ***@***.***>:

**Comment by Yeelbil** [comment]
No difference, put up two templates. Exit game, and when i start it they're both gone.

**Comment by Lethay** [comment]
You can see templates for other classes even if you're on (e.g.) a Warrior,
right? The reason I'm asking about this is because your settings file has
the flag active that comes up if Talented Data is/was missing, and *force
load* was set to False.

2021年7月25日(日) 17:38 Yeelbil ***@***.***>:

**Comment by Yeelbil** [comment]
Yes, if im on my mage i can see the template i created on my warrior.
